rusk/src/lib/services/state.rs

```diff
@@ -37,30 +37,29 @@ pub use super::rusk_proto::{
 
 pub(crate) type SpentTransaction = (Transaction, GasMeter);
 
-use super::{TX_TYPE_COINBASE, TX_TYPE_TRANSFER, TX_VERSION};
-/// Partition transactions into transfer and coinbase transactions, in this
-/// order.
+use super::TX_TYPE_COINBASE;
+
+/// Partition transactions into transfer and coinbase notes.
 fn extract_coinbase(
-    tx: Vec<TransactionProto>,
+    mut txs: Vec<TransactionProto>,
 ) -> Result<(Vec<TransactionProto>, (Note, Note)), Status> {
-    let (transfer_txs, coinbase_txs): (Vec<_>, Vec<_>) =
-        tx.into_iter().partition(|tx| tx.r#type == TX_TYPE_TRANSFER);
-
-    // There must always be two Coinbase transactions
-    let coinbases = coinbase_txs.len();
-    if coinbases != 2 {
-        return Err(Status::invalid_argument(format!(
-            "Expected 2 coinbase transactions, got {}",
-            coinbases
-        )));
+    let coinbase_tx = txs.pop().ok_or_else(|| {
+        Status::invalid_argument("Expected at least one transaction")
+    })?;
```

Comment:
I would still prefer to filter it out, unless we're absolutely sure no implementation (not only ours) from protocol is able to put the coinbase in a different position than last.

Reply:
In the conversations I had with @autholykos, I remember arriving at a situation where we were deciding on whether the coinbase should always be last or if it could be any other position in the block. I think we decided in favor of having it always be last. Can you confirm this @autholykos? Maybe we should document this as a part of the protocol.

Reply:
The position of the `Coinbase` was chosen to be last for no other reason but preventing the Block Generator to spend the Coinbase in the same block. I don't have an immediate reason for preventing this other than an excess of cautiousness. So, no, there is no reason I can come up with that this should be formally part of the protocol

Reply:
So as we also clarified during the sync up, it would be preferable to go back to filtering this out since that detail is currently not part of any protocol.